pkg/sessionctx/variable/session.go

```diff
@@ -203,11 +204,6 @@ type TxnCtxNoNeedToRestore struct {
 	StartTS     uint64
 	StaleReadTs uint64
 
-	// ShardStep indicates the max size of continuous rowid shard in one transaction.
-	ShardStep    int
-	shardRemain  int
-	currentShard int64
```

Comment:
This PR removed those lines because all states are moved to `SessionVars`. And every time the txn active, the inner shardStep and shardRemain will be reset